feat(datepicker): improve default look and feel

maxokorokov opened the pull request:
Updating default datepicker look and feel as a part of #862 :

* Moving to flex layout instead of table layout
* More space and border around the datepicker
* Less pronounced weekdays
* Visual separation of the header from the body
* Buttons same as in the timepicker
* Prefixing css classes with `ngb-dp-`

Before:
![screen shot 2017-01-10 at 11 38 46](https://cloud.githubusercontent.com/assets/8074436/21803211/600fd3c4-d729-11e6-9015-6049a7be2951.png)

After:
![screen shot 2017-01-10 at 11 39 17](https://cloud.githubusercontent.com/assets/8074436/21803229/7223732c-d729-11e6-9cdf-9b9be4875c3e.png)

![screen shot 2017-01-10 at 11 41 47](https://cloud.githubusercontent.com/assets/8074436/21803309/ce39295e-d729-11e6-9b99-6cdb03a94aa2.png)

P.S. Will also add full month names with a separate request later

BREAKING CHANGES: switched to flex layout instead of table-based (drop IE9 support) and prefixed datepicker-related css classes with `ngb-dp-`

Fixes #706

deeg commented on src/datepicker/datepicker-day-view.ts:
```diff
@@ -4,16 +4,22 @@ import {NgbDateStruct} from './ngb-date-struct';
     '[class.bg-primary]': 'selected',
     '[class.text-white]': 'selected',
     '[class.text-muted]': 'isMuted()',
+    '[class.outside]': 'isMuted()',
```
What is the reason for the different classes if they go off the same exact check? Is it possible to have text-muted without outside, or vice versa?

maxokorokov replied:
`text-muted` is the bootstrap class. I wanted to add 50% opacity on top of it, so two options:

- `[style.opacity]="isMuted() ? 0.5 : 1"`
- add the css class

I picked the last one, not sure which one is better though

deeg commented on src/datepicker/datepicker.ts:
```diff
@@ -47,36 +47,63 @@ export interface NgbDatepickerNavigateEvent {
+    <div class="ngb-dp-navigation bg-faded">
+      <ngb-datepicker-navigation *ngIf="navigation !== 'none'"
+        [date]="months[0]?.firstDate"
+        [minDate]="_minDate"
```
Should private variables be used in templates? Not sure if these should be public or not. Maybe they are, just going by the name of the var here.

maxokorokov replied:
Yeah, this is not a private variable... the thing is there is an `@Input() minDate: NgbDateStruct` and an internal data model `_minDate: NgbDate`

Not sure about this whole `NgbDateStruct` vs `NgbDate` thing, but it's a whole another story :)

deeg commented on src/datepicker/datepicker.ts:
```diff
@@ -47,36 +47,63 @@ export interface NgbDatepickerNavigateEvent {
+    </div>
+
+    <div class="ngb-dp-months d-flex">
+      <div class="ngb-dp-month-spacer-sm bg-faded" [style.height.rem]="navigation !== 'select' || displayMonths > 1 ? 4 : 2"></div>
```
I don't mind it that much, but might it be easier to read if you move this logic to a function? Or is the standard to keep simple stuff like this int he HTML?

maxokorokov replied:
I've just simplified layout significantly and got rid of spacers altogether. Done with css now. Thanks for pointing this

maxokorokov commented:
Amended:

- simplified layout even more → removed spacers and did them with css
- fixed and added tests with various `[showWeekdays]`, `[navigation]` and `[displayMonths]` values
- fixed TS < 2.0 compatibility

P.S. Tested in IE 10, 11, Edge. Seems to look fine.

Matmo10 commented:
Hi @maxokorokov, thanks for the visual refresh on this.   Do you happen to know what version this will land / landed in? I just installed `1.0.0-alpha.20` but I still see the old calendar. I'm just slightly confused as the demo website already shows the new design, and says that `1.0.0-alpha.20` is the latest release - but I can't see the new calendar after installing `1.0.0-alpha.20`.

maxokorokov commented:
@Matmo10 it landed in [alpha.19](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CHANGELOG.md#100-alpha19-2017-01-25).

I've just upgraded to `alpha.20` on one of my projects and it worked fine. 
Did the same, also deleted `node_modules/@ng-bootstrap` before installing just to be sure too.
